x/jobs/block_header_job.go

```diff
@@ -0,0 +1,40 @@
+package jobs
+
+import (
+	"context"
+
+	"github.com/berachain/offchain-sdk/job"
+	sdk "github.com/berachain/offchain-sdk/types"
+	"github.com/ethereum/go-ethereum"
+	coretypes "github.com/ethereum/go-ethereum/core/types"
+)
+
+// Compile time check to ensure that EthEventSub implements job.BlockHeaderSub.
+var _ job.BlockHeaderSub = (*BlockHeaderWatcher)(nil)
+
+// BlockHeaderWatcher allows you to subscribe a basic job to a block header event.
+type BlockHeaderWatcher struct {
+	job.Basic
+}
+
+// NewBlockHeaderWatcher creates a new BlockHeaderWatcher.
+func NewBlockHeaderWatcher(basic job.Basic) *BlockHeaderWatcher {
+	return &BlockHeaderWatcher{
+		Basic: basic,
+	}
+}
+
+func (w *BlockHeaderWatcher) Subscribe(ctx context.Context) (ethereum.Subscription, chan *coretypes.Header) {
```

Comment:
Personally, I think we should kill Subscribable jobs and just do this in Setup() and Teardown() like we have for other jobs.

Reply:
agree, but don't have bandwidth to do atm